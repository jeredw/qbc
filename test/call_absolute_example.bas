    'Calls routine for printing the screen to a local printer.
    DIM a%(2)
    DEF SEG = VARSEG(a%(0))
    FOR i% = 0 TO 2
       READ d%
       POKE VARPTR(a%(0)) + i%, d%
    NEXT i%
    DATA 205, 5, 203  : ' int 5  retf  'Machine-language code
                                       'for printing screen.
    CALL ABSOLUTE(VARPTR(a%(0)))
    DEF SEG
