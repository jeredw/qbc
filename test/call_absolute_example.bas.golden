(program statement \n 
  (statement 
    (dim_statement DIM 
      (dim_variable a% 
        (dim_array_bounds 
          ( 
            (dim_subscript 
              (expr 
                (literal 2))) ))))) \n 
  (statement 
    (def_seg_statement DEF SEG = 
      (expr 
        (variable_or_function_call VARSEG 
          (args_or_indices 
            ( 
              (expr 
                (variable_or_function_call a% 
                  (args_or_indices 
                    ( 
                      (expr 
                        (literal 0)) )))) )))))) \n 
  (statement 
    (for_next_statement FOR i% = 
      (expr 
        (literal 0)) TO 
      (expr 
        (literal 2)) 
      (block \n 
        (statement 
          (read_statement READ 
            (read_argument_list 
              (variable_or_function_call d%)))) \n 
        (statement 
          (call_statement 
            (untyped_id POKE) 
            (call_argument_list 
              (call_argument 
                (expr 
                  (expr 
                    (variable_or_function_call VARPTR 
                      (args_or_indices 
                        ( 
                          (expr 
                            (variable_or_function_call a% 
                              (args_or_indices 
                                ( 
                                  (expr 
                                    (literal 0)) )))) )))) + 
                  (expr 
                    (variable_or_function_call i%)))) , 
              (call_argument d%)))) \n statement) NEXT i%)) \n 
  (statement 
    (call_statement 
      (untyped_id DATA) 
      (call_argument_list 
        (call_argument 
          (expr 
            (literal 205))) , 
        (call_argument 
          (expr 
            (literal 5))) , 
        (call_argument 
          (expr 
            (literal 203)))))) : statement \n statement \n 
  (statement 
    (call_absolute_statement CALL ABSOLUTE 
      ( 
        (expr 
          (variable_or_function_call VARPTR 
            (args_or_indices 
              ( 
                (expr 
                  (variable_or_function_call a% 
                    (args_or_indices 
                      ( 
                        (expr 
                          (literal 0)) )))) )))) ))) \n 
  (statement 
    (def_seg_statement DEF SEG)) \n)
